using-directus-as-a-baby-health-tracker/index.md

```diff
@@ -0,0 +1,362 @@
+The general process to get a token which can be used to access actual device data contains three steps:
+
+1. First, we use the regular login credentials (the one I can log in within the mobile App) and pass them to an auth service provided by Google. If successfully logged in, we receive a first token.
+2. With this token we can now secondly authenticate against the actual vendor server. But again we just get a second token.
+3. Within the third request we pass the 2nd token to the vendor API once again. For this request, we will receive a last token that can be used to fetch actual data from the API.
+
+!(Pasted image 20240214224605.png)
+
+The screenshot shows the last request. As a key, we've set `token_sign_in` which can be used in the next operation. Within the payload/request body, we've specified the token to use with the dynamic code `"{{$last.data.mini_token}}"`. The object `$last` refers to the previous operation, an easy way to access the results from the last request. 
+
+!(Pasted image 20240216105507.png)
+ 
+##### Getting actual user data
+Compared to the auth process, fetching actual data from the API is done with fewer requests – just two are needed. As the vendor supports multiple devices, we have to get a list of devices first. In theory, we can also do this in preparation and hard-code the device ID within the next operation. But again, as there is no official documentation of the API this might change or the ID (which is more of a name) might not that unique and may change within any future updates. 
+
+So let's catch the device name, or `dsn` as it is called by the external API. Again we are using a Web request with the last token we got from the auth-process.
+
+The GET URL is static and does not contain any dynamic value this time.
+
+| Method | URL |
+| --- | --- |
+| GET | https://ads-field-eu-1a2039d9.aylanetworks.com/apiv1/devices.json |
+As header parameter, we set the following:
+
+| Header | Value |
+| --- | --- |
+| Authorization | `auth_token {{$last.data.access_token}}` |
+As you can see, we use the `{{ … }}` to pass the access_token from the last auth request.
+
+If everything works so far, the JSON we receive looks something like this, but for security reasons, some data was removed or masked as it also contains coordinates, MAC addresses, IP addresses and so on:
+```json
+{
+  "status": 200,
+  "statusText": "OK",
+  "headers": {
+    ...
+  },
+  "data": [
+    {
+      "device": {
+        "dsn": "AC00***697",
+        "lan_ip": "192.168.*.*",
+        "lan_enabled": false,
+        "has_properties": true,
+        "connection_status": "Online",
+        "lat": "********",
+        "lng": "********",
+        "device_type": "Wifi"
+      }
+    }
+  ]
+}
+```
+
+The device object has exactly one tracker and provides us with the `dsn` as well as the flag for `has_properties`. With the next operation, we will request all of these properties to extract the sensor data from the tracker.
```

Comment:
````suggestion
## Reading Data From the Sensor

Compared to the auth process, fetching actual data from the API is done with fewer requests – just two are needed. As the vendor supports multiple devices, we have to get a list of devices first. In theory, we can also do this in preparation and hard-code the device ID within the next operation. But again, as there is no official documentation of the API this might change or the ID might not that unique and may change within any future updates. 

So let's identify the device name, or `dsn` as it is called by the external API. 

Create a Request operation with the final token we got from the authentication operations.


| Method | URL |
| --- | --- |
| GET | https://ads-field-eu-1a2039d9.aylanetworks.com/apiv1/devices.json |

As header parameter, we set the following:

| Header | Value |
| --- | --- |
| Authorization | `auth_token {{$last.data.access_token}}` |

If everything works so far, the JSON we receive looks something like this, but for security reasons, some data was removed or masked as it also contains coordinates, MAC addresses, IP addresses, and so on:

```json
{
  "status": 200,
  "statusText": "OK",
  "headers": {
    ...
  },
  "data": [
    {
      "device": {
        "dsn": "AC00***697",
        "lan_ip": "192.168.*.*",
        "lan_enabled": false,
        "has_properties": true,
        "connection_status": "Online",
        "lat": "********",
        "lng": "********",
        "device_type": "Wifi"
      }
    }
  ]
}
```

The device object has exactly one tracker and provides us with the `dsn` as well as the flag for `has_properties`.
````